File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        // using hashing
        int n = nums.size();
        
        vector<int> ans;
        
        map<int, int> mpp;
        // key difference and value index
        
        for(int i = 0; i < n; i++) {
            int diff = target - nums[i];
            
            if(mpp.find(diff) != mpp.end()) {
                ans.push_back(i);
                ans.push_back(mpp[diff]);
                break;
            }
            
            mpp[nums[i]] = i;
         }
        
        return ans;
    }
};

File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    m := make(map[int]int)
    
    for i1, num := range nums {
        if i2, found := m[target - num]; found {
            return []int{i1, i2}
        }
        
        m[num] = i1;
    }
    
    return nil
}